Torchserve Changes for kserve migration

jagadeeshi2i opened the pull request:
**What this PR does / why we need it**:
Changes for KFServing to KServe Migration in Torchserve.

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
deprecate `service_envelope` in config.properties and use `enable_envvars_config: true` to enable service envelop at runtime.
```

Commit:
feat: v2 protocol support for torchserve

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add unit test for torchserve predictor

 - Add e2e test

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

feat: Add custom handlers for v2 api

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Update torchserve image in test overlay

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Update test_transformer.py

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Add e2e test for grpc torchserve

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Update test configuration

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Support torchserve runtime for v2 protocol

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Fix storage uri for v2 example

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Add grpc debug and retry config

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

fix: torchserve gRPC test

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

Commit:
feat: v2 protocol support for torchserve

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add unit test for torchserve predictor

 - Add e2e test

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

feat: Add custom handlers for v2 api

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Update torchserve image in test overlay

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Update test_transformer.py

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Add e2e test for grpc torchserve

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Update test configuration

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Support torchserve runtime for v2 protocol

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Fix storage uri for v2 example

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

Add grpc debug and retry config

Signed-off-by: Dan Sun <dsun20@bloomberg.net>

fix: torchserve gRPC test

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

Commit:
Feat: Changes on Torchserve repo for KFServing to Kserve migration

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

Commit:
Feat: Changes on Torchserve repo for KFServing to Kserve migration

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

Commit:
Merge remote-tracking branch 'upstrem/master' into torchserve-kserve

yuzisun commented on pkg/apis/serving/v1beta1/predictor_torchserve_test.go:
```diff
@@ -307,7 +307,7 @@ func TestCreateTorchServeModelServingContainer(t *testing.T) {
 					v1.EnvVar{
 						Name:  "TS_SERVICE_ENVELOPE",
-						Value: "kfserving",
+						Value: "kserve",
```
Let's use the constant here

Commit:
Merge remote-tracking branch 'origin/torchserve-kserve' into torchserve-kserve

Suresh-Nakkeran committed:
update service envelope env variable in serving runtime

Signed-off-by: Suresh Nakkeran <suresh.n@ideas2it.com>

Commit:
Merge remote-tracking branch 'origin/torchserve-kserve' into torchserve-kserve

Commit:
fix: move envvar to constants

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

Commit:
Merge remote-tracking branch 'upstrem/master' into torchserve-kserve

jagadeeshi2i commented:
/retest

yuzisun commented:
@jagadeeshi2i Seems like it is crashing in the mutating webhook, @Suresh-Nakkeran might be able to help.
```
W
        /usr/local/go/src/net/http/server.go:1802 +0xb9
panic({0x184dec0, 0x1d24a00})
        /usr/local/go/src/runtime/panic.go:1047 +0x266
github.com/kserve/kserve/pkg/apis/serving/v1beta1.(*InferenceService).assignPyTorchRuntime(...)
        /go/src/github.com/kserve/kserve/pkg/apis/serving/v1beta1/inference_service_defaults.go:213
github.com/kserve/kserve/pkg/apis/serving/v1beta1.(*InferenceService).setPredictorModelDefaults(0xc000505000)
        /go/src/github.com/kserve/kserve/pkg/apis/serving/v1beta1/inference_service_defaults.go:106 +0x225
github.com/kserve/kserve/pkg/apis/serving/v1beta1.(*InferenceService).DefaultInferenceService(0xc000505000, 0xc00039c000)
        /go/src/github.com/kserve/kserve/pkg/apis/serving/v1beta1/inference_service_defaults.go:77 +0x38
github.com/kserve/kserve/pkg/apis/serving/v1beta1.(*InferenceService).Default(0xc000505000)
```

Suresh-Nakkeran committed:
fix: nil pointer error

Signed-off-by: Suresh Nakkeran <suresh.n@ideas2it.com>

jagadeeshi2i commented:
/retest

yuzisun commented:
/retest

jagadeeshi2i commented:
/retest

Commit:
fix: torchserve image with kserve envelope

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

jagadeeshi2i commented:
/retest

Commit:
fix: torchserve runtime image

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

jagadeeshi2i commented:
/retest

jagadeeshi2i commented:
/retest

jagadeeshi2i commented:
/retest

jagadeeshi2i commented:
/retest

jagadeeshi2i commented:
/retest

yuzisun commented:
/retest

jagadeeshi2i commented:
/retest

yuzisun commented on test/e2e/predictor/test_torchserve.py:
```diff
@@ -159,7 +159,7 @@ def test_torchserve_runtime_kserve():
             ),
-            env=[V1EnvVar(name="TS_SERVICE_ENVELOPE", value="kfserving")],
+            env=[V1EnvVar(name="TS_SERVICE_ENVELOPE", value="kserve")],
             storage_uri="gs://kfserving-examples/models/torchserve/image_classifier",
```
Please help update to `gs://kfserving-examples/models/torchserve/image_classifier/v1`

yuzisun commented on docs/samples/v1beta1/torchserve/README.md:
```diff
@@ -39,11 +39,9 @@ torch-model-archiver --model-name mnist --version 1.0 \
 - For PVC user please refer to [model archive file generation](./model-archiver/README.md) for auto generation of MAR files from
 the model and dependent files.
 
-- The [config.properties](./v1/config.properties) file includes the flag `service_envelope=kfserving` to enable the KServe inference protocol.
```
Can you add instruction here that `service_envelope` is deprecated and use `enable_envvars_config` instead.

yuzisun commented on config/configmap/inferenceservice.yaml:
```diff
@@ -61,9 +61,9 @@ data:
-          "image": "kserve/torchserve-kfs",
-          "defaultImageVersion": "0.5.0",
-          "defaultGpuImageVersion": "0.5.0-gpu",
+          "image": "jagadeeshj/torchserve-kfs",
```
```suggestion
          "image": "kserve/torchserve-kfs",
```

yuzisun commented on config/overlays/test/configmap/inferenceservice.yaml:
```diff
@@ -61,9 +61,9 @@ data:
-          "image": "kserve/torchserve-kfs",
-          "defaultImageVersion": "0.5.0",
-          "defaultGpuImageVersion": "0.5.0-gpu",
+          "image": "jagadeeshj/torchserve-kfs",
```
```suggestion
          "image": "kserve/torchserve-kfs",
```

yuzisun commented on config/runtimes/kustomization.yaml:
```diff
@@ -44,5 +44,5 @@ images:
   - name: kserve-torchserve
-    newName: kserve/torchserve-kfs
-    newTag: 0.5.0
+    newName: jagadeeshj/torchserve-kfs
```
```suggestion
    newName: kserve/torchserve-kfs
```

Commit:
fix: v1 example storage uri

 - add service_envelope deprecation warining to readme
 - update torchserve image

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

yuzisun commented:
@jagadeeshi2i there is another place missing in `test_raw_transformer.py`

Commit:
fix: test_raw_transformer.py

Signed-off-by: Jagadeesh J <jagadeeshj@ideas2it.com>

yuzisun commented:
/lgtm
/approve
